### PR TITLE
Only do out-of-order compatible verifications at verifier-thread time.

### LINKED ISSUE #2648: Don't do parent verification checks in verifier thread
verifier threads can be executed out of order, parentHashes may not be available in chain at he time of verification.

### PR DESCRIPTION
**Pull request by gavofyork**
Do other verifications at import time.
Fixed #2648

### CODE REVIEW

**Comment by gavofyork** [comment]
excuse the additional state dump stuff...

File: alethzero/MainWin.cpp [hunk 1]
```diff
@@ -1851,6 +1851,43 @@ void Main::on_debugCurrent_triggered()
...
 }
 
+std::string minHex(h256 const& _h)
+{
+	unsigned i = 0;
+	for (; i < 31 && !_h[i]; ++i) {}
+	return toHex(_h.ref().cropped(i));
+}
+
+void Main::on_dumpBlockState_triggered()
```
Comment (chriseth) [inline alethzero/MainWin.cpp]: Does this work without fatdb?

Comment (gavofyork) [inline alethzero/MainWin.cpp]: fixed.

**Comment by CJentzsch** [comment]
looks good

**Comment by CJentzsch** [comment]
all tests pass

**Pull request merged**